### PR TITLE
Add ability to specify global options in a test

### PR DESCRIPTION
**Pull request by jmarolf**
closes https://github.com/dotnet/roslyn-sdk/issues/823

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.SourceGenerators.Testing/SourceGeneratorTest`1.cs [hunk 1]
```diff
@@ -22,6 +20,11 @@ namespace Microsoft.CodeAnalysis.Testing
     public abstract class SourceGeneratorTest<TVerifier> : AnalyzerTest<TVerifier>
         where TVerifier : IVerifier, new()
     {
+        /// <summary>
+        /// Gets a collection of global options that will be awaiable from Context.AnalyzerConfigOptions.GlobalOptions
+        /// </summary>
+        public List<(string key, string value)> GlobalOptions { get; } = new List<(string, string)>();
```
Comment (sharwell) [inline src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.SourceGenerators.Testing/SourceGeneratorTest`1.cs:26]: This is already provided by TestState.AnalyzerConfigFiles.

**Review by sharwell — CHANGES_REQUESTED** [review]

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.SourceGenerators.Testing/SourceGeneratorTest`1.cs (hunk 1 above)
Comment (jmarolf) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.SourceGenerators.Testing/SourceGeneratorTest`1.cs:26]: what about msbuild files? would our approach then be to have the user supply the props/targets files?

Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.SourceGenerators.Testing/SourceGeneratorTest`1.cs:26]: No, the testing focuses on the compiler component of a build.

Comment (jmarolf) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.SourceGenerators.Testing/SourceGeneratorTest`1.cs:26]: we support adding nuget packages which is not a compiler component of a build. I don't think API is what we want but we don't think providing a way to test this makes sense?

Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.SourceGenerators.Testing/SourceGeneratorTest`1.cs:26]: I don't understand the request here. We already have the ability to test global options through an existing API. Why would we need to add a new one?

**Pull request closed without merging**